chain/chain/src/store.rs

```diff
@@ -1906,7 +1906,9 @@ impl<'a> ChainStoreUpdate<'a> {
 
     pub fn update_tail(&mut self, height: BlockHeight) {
         self.tail = Some(height);
-        if self.chunk_tail.is_none() {
+        let genesis_height = self.get_genesis_height();
+        let chunk_tail = self.chunk_tail().unwrap_or_else(|_| genesis_height);
+        if chunk_tail == genesis_height {
             // For consistency, Chunk Tail should be set if Tail is set
             self.chunk_tail = Some(self.get_genesis_height());
         }
```

Comment:
```suggestion
        // For consistency, Chunk Tail should be set if Tail is set
        self.update_chunk_tail(self.chunk_tail().unwrap_or_else(|_| self.get_genesis_height()));
```

Reply:
@Kouprin I was going to do this, but seems like this will cause unnecessary writes to storage if chunk tail exists (which would be the normal case).

Reply:
You mean storage reading, right?
Anyway, we call `update_tail` only in GC once per each height. In average, we do lots of storage interventions per height in GC. I'd prefer to sacrifice small optimization for code quality here.

Reply:
> You mean storage reading, right?

No, you call `chunk_tail()`, so there is already reading from storage. Sorry, I can't understand what you're talking about.

Reply:
I was thinking about the storage write to `CHUNK_TAIL` when we commit the update, but I think you are right about optimization vs. readability. However, the PR is already merged :(

Reply:
> I was thinking about the storage write to `CHUNK_TAIL` when we commit the update, but I think you are right about optimization vs. readability. However, the PR is already merged :(

There is no optimization because we write `chunk_tail` anyway.

Reply:
We only write it if chunk_tail is not `None`.